### ISSUE TITLE
Filtering trials and/or searching for them and adding multiple columns to trial table

### ISSUE THREAD

**Issue by sylvr3** [ser515asu/PRP-Manhattan-Project#68]
So I am looking at how the trial completion status should be filtered by using the dropdown menu on the dashboard page. 

Do we only want to filter the trials by Active or Complete? How do we determine whether a trial is in the Pending or Upcoming state (as seen in the dropdown menu options)? I am assuming that Upcoming means that it has been scheduled ahead of time, but I am not sure about Pending.

Also, I was thinking of adding a search box that allows the clinician to search for a particular trial by any of the columns (not just the trial name). Is a search box a good enhancement or is that not necessary in our case?

As far as filtering with the dropdown menu goes, I think sorting the trial table in its current state will be difficult because there are multiple line breaks and values defined in columns 2-4. I understand that having several columns for each trial attribute would make the table cluttered, but it would making sorting the table much easier for me. 

Thoughts? @beccannlittle

**Comment by ChristianMurphy** [comment]
> How do we determine whether a trial is in the Pending or Upcoming state

**Pending:** Don't worry about that right now, that won't be in the MVP
**Upcoming:** IRB Start date is in the future
**In Progress:** current date is after IRB start date and before IRB end date
**Completed:** IRB end date has passed

> Is a search box a good enhancement or is that not necessary in our case?

It would be nice, but is not necessary.
IMO hold off on doing it.

> As far as filtering with the dropdown menu goes, I think sorting the trial table in its current state will be difficult because there are multiple line breaks and values defined in columns 2-4. I understand that having several columns for each trial attribute would make the table cluttered, but it would making sorting the table much easier for me.

Sorting will always be off a single property.
IMHO only name, IRB start and IRB end need to be sortable.

**Comment by sylvr3** [comment]
Do you recommend using the jQuery plugin tablesorter? 
https://github.com/Mottie/tablesorter

**Comment by ChristianMurphy** [comment]
That depends on if we do pagination or not.
If we do pagination, then no purely front end solution would work.
If we don't do pagination, then tablesorter would be a good pick.

@beccannlittle thoughts on pagination?

**Comment by beccannlittle** [comment]
There are plenty of tools that allow for client-side pagination. For example, [DataTables](https://www.datatables.net/) is one I've used before that plays well with Bootstrap and has tons of other features, like filtering, sorting, search, etc.

**Comment by beccannlittle** [comment]
The search functionality would also help with the sorting issue we have with our complex cells, since it indexes all text in the table.

**Comment by sylvr3** [comment]
How many trials do we want to show per page? I was thinking 25.

**Comment by beccannlittle** [comment]
That's a shitton haha. 10 is a pretty standard amount for pagination.

**Comment by sylvr3** [comment]
:+1: Since we're using the search functionality and filtering in DataTables, can we get rid of the dropdown menu?

**Comment by beccannlittle** [comment]
Purpose of the drop down menu would be to separate distinct categories of trials, to remove inactive trials from view. For example, consider archived trials. Do you want them to exist in the table forever? I thought it would be more useful to default to hiding them. That way they're still accessible, but they're also out of the way.

**Comment by sylvr3** [comment]
Here is the issue that I am having if I try to sort the table using DataTables:

![screenshot from 2015-11-11 11 55 24](https://cloud.githubusercontent.com/assets/1154846/11099889/3c6aeae8-886b-11e5-95d5-56ba27b1a3fa.png)

![screenshot from 2015-11-11 11 52 26](https://cloud.githubusercontent.com/assets/1154846/11099898/4ab652ea-886b-11e5-8fc6-78537140120e.png)

The search functionality works fine, but dataTables is causing the tables to format in a weird way every time I add a new trial. :disappointed: 

This is happening because of the <td> tags that show the % Recruited, Active, etc.

What should we do? Should we add the recruitedPercent and other percentages into their own separate columns? Should we only allow the clinician to search for trials but not filter with the dropdown menu? 

Getting really frustrated... :angry: Any thoughts?

**Comment by ChristianMurphy** [comment]
> What should we do? Should we add the recruitedPercent and other percentages into their own separate columns? Should we only allow the clinician to search for trials but not filter with the dropdown menu?
> Getting really frustrated... :angry: Any thoughts?

My recommendation.
Look into using a custom rendering for these columns
https://datatables.net/reference/option/columns.render#Examples
(Examples 4 and 5 show to render tags with content)

**Comment by sylvr3** [comment]
Only problem with that approach is that JS Lint keeps throwing errors for the data sources and does not like the single quotes around the properties. (For example: Unnecessarily quoted property `columnDefs` found)
Is there a way to bypass that error?

**Comment by ChristianMurphy** [comment]
by removing the quotes around the properties.

``` js 
$('#example').dataTable( {
  "columnDefs": [ {
    "targets": 0,
    "data": "download_link",
    "render": function ( data, type, full, meta ) {
      return '<a href="'+data+'">Download</a>';
    }
  } ]
} );
```

is equivalent to
``` js 
$('#example').dataTable( {
  columnDefs: [ {
    targets: 0,
    data: "download_link",
    render: function ( data, type, full, meta ) {
      return '<a href="'+data+'">Download</a>';
    }
  } ]
} );
```

only difference is the second will not throw the lint error for `quote-props` :smile:

**Comment by sylvr3** [comment]
Resolved in pull request #77
